ons_alpha/core/blocks/markup.py

```diff
@@ -95,6 +96,17 @@ def _get_rows(self, value, classnames, hidden, spans):  # pylint: disable=too-ma
             trs.append({"tds": tds})
         return trs
 
+    def clean(self, value):
+        if not value or not value.get("table_header_choice"):
+            raise ValidationError("Select an option for Table headers")
+
+        data = value.get("data", [])
+        all_cells_empty = all(not cell for row in data for cell in row)
+        if all_cells_empty:
+            raise ValidationError("Table cannot be empty")
```

Comment:
small, non-blocking suggestion: 
```suggestion
            raise ValidationError("The table cannot be empty")
```